backend/data_preparation/dumper/fire_dumper.py

```diff
@@ -0,0 +1,438 @@
+    #     """
+    #     logger.info("Looking for fire table in database...")
+    #     # create a cursor
+    #     cur = conn.cursor()
+    #     # check if the fire_history table exists
+    #     # if the pipeline is run for the first time
+    #     # then the fire_history table will not exist
+    #     cur.execute(self.SQL_CHECK_IF_FIRE_TABLE_EXISTS)
+    #     tables = cur.fetchall()
+    #     # table is the result of the statement: sql_check_if_history_table_exists
+    #     if len(tables) == 0:
+    #         # if table is empty, which means the fire_history table does not exists
+    #         logger.info("No fire table exists. Creating a new one.")
+    #         cur.execute(FireDumper.SQL_CREATE_FIRE_TABLE)
+    #         # create the fire_history table
+    #         conn.commit()
+    #         # commit the transaction so the change will be saved
+    #         logger.info("Fire table created.")
+    #     else:
+    #         logger.info("Find the fire table, continue >..")
+    #     cur.close()
+    #
+    # def _create_fire_merged_table(self, conn: Connection):
+    #     """
+    #     Checks if the fire_merged table exists,
+    #     Creates fire_merged table if not exist
+    #     If it exists, the function does nothing
+    #     :param conn: Connection object
+    #     """
+    #     logger.info("Looking for fire_merged table in database...")
+    #     # create a cursor
+    #     cur = conn.cursor()
+    #     # check if the fire_merged table exists
+    #     # if the pipeline is run for the first time
+    #     # then the fire_merged table will not exist
+    #     # table is the result of the statement: sql_check_if_history_table_exists
+    #     if len(list(Connection.sql_execute(self.SQL_CHECK_IF_FIRE_MERGED_TABLE_EXISTS))) == 0:
+    #         # if table is empty, which means the fire_history table does not exists
+    #         logger.info("No fire_merged table exists. Creating a new one.")
+    #         cur.execute(FireDumper.SQL_CREATE_HISTORY_TABLE)
+    #         # create the fire_history table
+    #         conn.commit()
+    #         # commit the transaction so the change will be saved
+    #         logger.info("fire_merged table created.")
+    #     else:
+    #         logger.info("Find the fire_merged table, continue >..")
+    #     cur.close()
+
+    @staticmethod
+    def retrieve_all_fires() -> Set[FireEvent]:
+        """
+        Retrieves all fires in the database.
+        :return: set of FireEvent objects
+                 e.g. [FireEvent(-1, 2015, 'California', 'FireA'), FireEvent(-1, 2015, 'California', 'FireB')]
+        """
+        # check if the fire_history table exists
+        # if not exist, executing sql_retrieve_all_fires will return an error
+        FireDumper._create_table("fire_history")
+        # retrieve all fires in fire_history
+        set_of_fire_event_objects = list(map(lambda fire_event_tuple: FireEvent.from_tuple(fire_event_tuple),
```

Comment:
Usually fetching all data from database is not a good idea. I assume here the data fetched is relatively small?